pkg/plugin/aggregation/run.go

```diff
@@ -154,3 +167,32 @@ func Cleanup(client kubernetes.Interface, plugins []plugin.Interface) {
 		p.Cleanup(client)
 	}
 }
+
+func sendUpdates(updater *status.Updater, ag *Aggregator) {
+	// Could have race conditions, but will be eventually consistent
+	for _, result := range ag.Results {
+		state := "complete"
+		if result.Error != "" {
+			state = "failed"
+		}
+		update := status.Plugin{
+			Node:   result.NodeName,
+			Plugin: result.ResultType,
+			Status: state,
+		}
+
+		if err := updater.Receive(&update); err != nil {
+			logrus.WithFields(
+				logrus.Fields{
+					"node":   update.Node,
+					"plugin": update.Plugin,
+					"status": state,
+					"at":     "sendUpdates",
```

Comment:
If the location of the failure is important we could include the error stacktrace using `pkg/errors`: https://godoc.org/github.com/pkg/errors#hdr-Retrieving_the_stack_trace_of_an_error_or_wrapper

Reply:
this is an old habit from Heroku's coding style. I can ditch it.